src/Widget/FormBuilder.php

```diff
@@ -0,0 +1,558 @@
+            ->start();
+        $html .= $content;
+        $html .= Forms::end();
+
+        return $html;
+    }
+
+    /**
+     * Generates a summary of the validation errors.
+     *
+     * If there is no validation error, an empty error summary markup will still be generated, but it will be hidden.
+     *
+     * @param FormInterface $forms the forms(s) associated with this form.
+     * @param array $options the tag options in terms of name-value pairs. The following options are specially handled:
+     *
+     * - `header`: string, the header HTML for the error summary. If not set, a default prompt string will be used.
+     * - `footer`: string, the footer HTML for the error summary.
+     *
+     * The rest of the options will be rendered as the attributes of the container tag. The values will be HTML-encoded
+     * using {@see \Yiisoft\Html\Html::encode()}. If a value is `null`, the corresponding attribute will not be
+     * rendered.
+     *
+     * @return string the generated error summary.
+     *
+     * @throws InvalidConfigException
+     *
+     * {@see errorSummaryCssClass}
+     */
+    public function errorSummary(FormInterface $forms, array $options = []): string
+    {
+        Html::addCssClass($options, $this->errorSummaryCssClass);
+
+        $options['encode'] = $this->encodeErrorSummary;
+
+        return ErrorSummary::widget()->form($forms)->options($options)->run();
+    }
+
+    /**
+     * Validates one or several forms and returns an error message array indexed by the attribute IDs.
+     *
+     * This is a helper method that simplifies the way of writing AJAX validation code.
+     *
+     * For example, you may use the following code in a controller action to respond to an AJAX validation request:
+     *
+     * ```php
+     * // ... respond to non-AJAX request ...
+     * ```
+     *
+     * To validate multiple forms, simply pass each model as a parameter to this method, like the following:
+     *
+     * ```php
+     * ```
+     *
+     * @param FormInterface $form the form to be validated.
+     * @param string|null $attributes list of attributes that should be validated. If this parameter is empty, it means
+     * any attribute listed in the applicable validation rules should be validated.
+     *
+     * When this method is used to validate multiple forms, this parameter will be interpreted as a form.
+     *
+     * @return array the error message array indexed by the attribute IDs.
+     */
+    public static function validate(FormInterface $form, string $attributes = null): array
+    {
+        $result = [];
+        if ($attributes instanceof FormInterface) {
+            /** validating multiple forms */
+            $forms = func_get_args();
+            $attributes = null;
+        } else {
+            $forms = [$form];
+        }
+
+        /** @var FormInterface $form */
+        foreach ($forms as $form) {
+            $form->validate($attributes);
```

Comment:
validate() doesn't accept attribute names.

Reply:
the name of the attribute comes in the rules.

protected function rules(): array
{
    return [
        'login' => $this->loginRules(),
        'password' => $this->passwordRules()
    ];
}